### PR TITLE
switch to webfinger.js

### PR DESCRIPTION
**Pull request by michielbdejong**
I got it working, but I'm not sure whether src/oauth.js has actually become in any way simpler by using this...

### CODE REVIEW

File: src/oauth.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ function getClientId(platform) {
   }
 }
 ret.oauth = function(platform, userAddress, scopes) {
-  if (platform !== 'remotestorage') {
+  if (platform !== 'remotestorage' && platform !== 'remotestorage-allow-insecure-webfinger') {
```
Comment (michielbdejong) [inline src/oauth.js:37]: this change is actually unrelated, btw.

File: src/oauth.js [hunk 2]
```diff
@@ -56,29 +56,29 @@ ret.oauth = function(platform, userAddress, scopes) {
...
     goTo('https://accounts.google.com/o/oauth2/auth');
-  } else if (platform === 'remotestorage' || platform === 'remotestorage-no-https') {
-    var parts = userAddress.split('@');
-    var prefix = 'https://';
-    if (platform === 'remotestorage-no-https') {
-      prefix = 'http://';
-    }
-    requestJSON(prefix + parts[1] + '/.well-known/webfinger?resource='+encodeURIComponent('acct:'+userAddress),
-        undefined, function(err, data) {
+  } else if (platform === 'remotestorage' || platform === 'remotestorage-allow-insecure-webfinger') {
```
Comment (michielbdejong) [inline src/oauth.js:59]: This one as well.

File: src/oauth.js [hunk 3]
```diff
@@ -56,29 +56,29 @@ ret.oauth = function(platform, userAddress, scopes) {
     goTo('https://www.dropbox.com/1/oauth2/authorize');
   } else if (platform === 'googledrive') {
     goTo('https://accounts.google.com/o/oauth2/auth');
-  } else if (platform === 'remotestorage' || platform === 'remotestorage-no-https') {
-    var parts = userAddress.split('@');
-    var prefix = 'https://';
-    if (platform === 'remotestorage-no-https') {
-      prefix = 'http://';
-    }
```
Comment (michielbdejong) [inline src/oauth.js:64]: These four lines are definitely good to get rid of.

File: src/oauth.js [hunk 4]
```diff
@@ -56,29 +56,29 @@ ret.oauth = function(platform, userAddress, scopes) {
...
-            apiBaseURL = data.links[i].href;
-            goTo(data.links[i].properties['http://tools.ietf.org/html/rfc6749#section-4.2']);
+      } else if ((typeof p.idx === 'object')
+          && (typeof p.idx.links == 'object')
+          && (Array.isArray(p.idx.links.remotestorage))) {
+        for (var i=0; i< p.idx.links.remotestorage.length; i++) {
+          if ((typeof p.idx.links.remotestorage[i] === 'object')
+              && typeof p.idx.links.remotestorage[i].href === 'string'
+              && typeof p.idx.links.remotestorage[i].properties === 'object'
+              && typeof p.idx.links.remotestorage[i].properties['http://tools.ietf.org/html/rfc6749#section-4.2'] === 'string') {
```
Comment (michielbdejong) [inline src/oauth.js]: Not sure if I like the whole `p.idx.links.remotestorage[i].properties` business. It seems far away from the actual data. Maybe I should have used `p.json` instead. But then, it would seem I'm not really using webfinger.js the way I'm supposed to, it would feel like short-circuiting

File: src/oauth.js [hunk 5]
```diff
@@ -56,29 +56,29 @@ ret.oauth = function(platform, userAddress, scopes) {
...
   } else if (platform === 'googledrive') {
     goTo('https://accounts.google.com/o/oauth2/auth');
-  } else if (platform === 'remotestorage' || platform === 'remotestorage-no-https') {
-    var parts = userAddress.split('@');
-    var prefix = 'https://';
-    if (platform === 'remotestorage-no-https') {
-      prefix = 'http://';
-    }
-    requestJSON(prefix + parts[1] + '/.well-known/webfinger?resource='+encodeURIComponent('acct:'+userAddress),
-        undefined, function(err, data) {
```
Comment (michielbdejong) [inline src/oauth.js:66]: I like how here you can see directly that `requestJSON` is about requesting JSON, and that the `data` variable is exactly the contents of the webfinger document. This simplicity makes everything easy to debug.

File: src/oauth.js (hunk 4 above)
Comment (silverbucket) [inline src/oauth.js]: You will always have `p.idx.links.remotestorage` as it's a known supported record type. It will always be an array. So you don't need to do any of this checking.

Comment (silverbucket) [inline src/oauth.js]: See here for a much simpler example:
https://github.com/remotestorage/remotestorage.js/blob/master/src/discover.js#L56-L60

Note that above this, there's a check for whether the remotestorage is an array or not. This is actually an unneeded check, since it'll always be an array you'd only care of the array is zero-length, then there's no remotestorage record.

Comment (michielbdejong) [inline src/oauth.js]: Ah ok, that's good to know. I removed the unnecessary checks (did leave them there as comments, to remind myself why they are not there)

File: ludbud.js [hunk 6]
```diff
@@ -243,30 +244,32 @@ ret.oauth = function(platform, userAddress, scopes) {
...
-              && typeof data.links[i].properties === 'object'
-              && typeof data.links[i].properties['http://tools.ietf.org/html/rfc6749#section-4.2'] === 'string') {
-            apiBaseURL = data.links[i].href;
-            goTo(data.links[i].properties['http://tools.ietf.org/html/rfc6749#section-4.2']);
+      } else {
+        //webfinger.js will now guarantee:
+        // (typeof p.idx === 'object') &&
+        // (typeof p.idx.links == 'object') &&
+        // (Array.isArray(p.idx.links.remotestorage))
+        for (var i=0; i< p.idx.links.remotestorage.length; i++) {
```
Comment (silverbucket) [inline ludbud.js]: Do you need to loop over this? You'll constantly overwrite the variables anyway. May as well just access the first record...

**Comment by silverbucket** [comment]
looks good to me!

**Comment by michielbdejong** [comment]
Added a PR to webfinger.js to make it the way I would want it to be :)

**Comment by michielbdejong** [comment]
Cleaning up PRs that are > 1 year old.
